### PR TITLE
[react-native-snap-carousel] Update definitions to match v3.x changes

### PR DESCRIPTION
**Pull request by j-fro**
`react-native-snap-carousel` v3.x includes several changes and additions to props (most significantly using `Flatlist` under the hood and so requiring a `data` and `renderItem` prop).

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: [README](https://github.com/archriss/react-native-snap-carousel)
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/react-native-snap-carousel/index.d.ts [hunk 1]
```diff
@@ -46,6 +57,16 @@ export interface CarouselProps extends React.Props<ScrollViewProperties> {
      * From slider's center, minimum slide distance to be scrolled before being set to active
      */
     activeSlideOffset?: number;
+    /**
+     * Duration of time while component is hidden after mounting. NOTE: May cause rendering
+     * issues on Android. Defaults to 0
+     */
+    apaparitionDelay?: number;
```
Comment (jnbt) [inline types/react-native-snap-carousel/index.d.ts]: This seems to be typo: `apparitionDelay`

File: types/react-native-snap-carousel/index.d.ts [hunk 2]
```diff
@@ -13,12 +14,22 @@ import {
...
 } from 'react-native';
 
-export interface CarouselProps extends React.Props<ScrollViewProperties> {
+export interface CarouselProps<T> extends React.Props<ScrollViewProperties> {
     // Required
 
+    /**
+     * Array of items to loop over
+     */
+    data?: ReadonlyArray<T>;
```
Comment (jnbt) [inline types/react-native-snap-carousel/index.d.ts]: I suspect this property to be required

File: types/react-native-snap-carousel/index.d.ts [hunk 3]
```diff
@@ -13,12 +14,22 @@ import {
...
+    /**
+     * Function that takes an item from the `data` array and returns a React
+     * Element. See `react-native`'s `FlatList`
+     */
+    renderItem?(item: { item: T; index: number }, index: number): React.ReactNode;
```
Comment (jnbt) [inline types/react-native-snap-carousel/index.d.ts]: I suspect this property to be required

File: types/react-native-snap-carousel/index.d.ts [hunk 4]
```diff
@@ -172,7 +230,38 @@ export interface CarouselStatic extends React.ComponentClass<CarouselProps> {
     snapToPrev(animated?: boolean): void;
 }
 
-export type CarouselProperties = ScrollViewProperties & CarouselProps & React.Props<CarouselStatic>;
+export type CarouselProperties<T> = ScrollViewProperties & CarouselProps<T> & React.Props<CarouselStatic<T>>;
```
Comment (jnbt) [inline types/react-native-snap-carousel/index.d.ts]: Could you add an example / test for the ParallaxImage to `react-native-snap-carousel-tests.tsx`?

**Review by jnbt — COMMENTED** [review]
@j-fro: Thanks for your work 💖  Please have a look into my comments.

**Review by jnbt — CHANGES_REQUESTED** [review]
Sorry didn't set to "Request changes"

File: types/react-native-snap-carousel/index.d.ts (hunk 2 above)
Comment (j-fro) [reply types/react-native-snap-carousel/index.d.ts]: Yes, thanks!

File: types/react-native-snap-carousel/index.d.ts (hunk 1 above)
Comment (j-fro) [reply types/react-native-snap-carousel/index.d.ts]: Thanks for catching!

File: types/react-native-snap-carousel/index.d.ts (hunk 4 above)
Comment (j-fro) [reply types/react-native-snap-carousel/index.d.ts]: Added a `SnapCarouselWithParallaxTest` example

**Review by jnbt — APPROVED** [review]
Thanks LGTM 👌🏿

**Pull request merged**